src/promptflow/setup.py

```diff
@@ -49,6 +49,10 @@
     "pyarrow>=14.0.1,<15.0.0",  # used to read parquet file with pandas.read_parquet
     "pillow>=10.1.0,<11.0.0",  # used to generate icon data URI for package tool
     "filetype>=1.2.0",  # used to detect the mime type for mulitmedia input
+    # otel dependencies for monitoring
+    "opentelemetry-api~=1.19.0",
+    "opentelemetry-sdk~=1.19.0",
+    "opentelemetry-exporter-prometheus>=0.42b0,<2.0.0",
 ]
```

Comment:
Maybe put this as extra requirements. 
If 3p user can benifit from this, we might add a new extra called serving. 
If this only benefites cloud user, we could put this under azure.

Reply:
so far we don't need extra requirement (we will need that till the new monitoring sdk come out)
this will benifit 3p customer, besides this default azure monitor exporter, we need time to better think about how to allow cutomer configuring their own exporter,  which will come in following PR(s)

Reply:
Let's add a monitoring extra as we offline discussed.

Reply:
updated